 alignment problems on SPARC. Based on similar fixes found at:

  https://lists.debian.org/debian-sparc/2012/01/msg00012.html

and:

  https://www.winehq.org/pipermail/wine-patches/2011-February/098547.html

This change needs to be sent upstream.

--- nmap-7.80/nping/nping.cc
+++ nmap-7.80/nping/nping.cc
@@ -173,6 +173,14 @@ int main(int argc, char *argv[] ){
   ProbeMode prob;
   NpingTarget *t=NULL;
 
+/*
+ * 'ta 6' tells the kernel to synthesize any unaligned accesses this process
+ * makes, instead of just signalling an error and terminating the process.
+ */
+#ifdef __sparc
+    __asm("ta 6");
+#endif
+
   /* Get current time */
   now = time(NULL);
   tm = localtime(&now);
